### PR TITLE
MINOR: Fix state transition diagram for stream threads

### PR DESCRIPTION
**Pull request by cadonna**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java [hunk 1]
```diff
@@ -93,7 +95,7 @@
      *          |      +-----+-------+      |
      *          +<---- | Partitions  |      |
      *          |      | Assigned (3)| <----+
-     *          |      +-----+-------+      |
+     *          |      +-----+-------+      ^
```
Comment (cadonna) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java]: Before this change a transition from `STARTING` to `RUNNING` was possible in the diagram. However, such a transition is not valid.

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java [hunk 2]
```diff
@@ -80,11 +80,13 @@
...
-     *          |            v              |
-     *          |      +-----+-------+      |
-     *          +<---- | Partitions  |      |
+     *          |                           |
+     *          |                           |
+     *          |            +<----------+  |
+     *          |            |           |  |
+     *          |            v           |  |
+     *          |      +-----+-------+   |  |
+     *          +<---- | Partitions  | --+  |
```
Comment (cadonna) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java]: This valid loop was only mentioned in the notes but was not shown in the diagram.

**Comment by cadonna** [comment]
Call for review: @guozhangwang @ableegoldman

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java (hunk 1 above)
Comment (ableegoldman) [reply streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java]: This seems a bit subtle. What if we instead break up the loop from Running --> Running below?

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java [hunk 3]
```diff
@@ -93,7 +95,7 @@
      *          |      +-----+-------+      |
      *          +<---- | Partitions  |      |
      *          |      | Assigned (3)| <----+
-     *          |      +-----+-------+      |
+     *          |      +-----+-------+      ^
      *          |            |              |
      *          |            |--------------+
```
Comment (ableegoldman) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java]: It seems like this here is the problem...AFAICT this line is just to demonstrate that RUNNING -> RUNNING is valid, but it's tied in to the arrows connecting every other state that actually can't transition to Running. Can we change this to a self-contained loop like you added for PARTITIONS_REVOKED?

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java (hunk 2 above)
Comment (ableegoldman) [reply streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java]: Nice catch

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java (hunk 1 above)
Comment (cadonna) [reply streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java]: I had the same feeling. I will change it.

**Review by ableegoldman — APPROVED** [review]
Thanks @cadonna ! LGTM

**Review by guozhangwang — APPROVED** [review]
LGTM

**Pull request merged**